src/Test/L1/Mock/FakeAgentPluginManager.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.L1.Worker
+{
+    public class FakeAgentPluginManager : AgentPluginManager
+    {
+        public override void Initialize(IHostContext hostContext)
+        {
+            // Inject any plugin mocks here.
+            // Each injection should be paired with a removal of the plugin being mocked.
```

Comment:
Should we add a private method here like `ReplacePlugin` that requires parameters to make sure people do it correctly?

Reply:
Sure, good idea - done